src/org/pentaho/agilebi/modeler/models/annotations/AnnotationType.java

```
@@ -359,6 +375,169 @@ public void populate( final Map<String, Serializable> propertiesMap ) {
+      OlapCube olapCube = null;
+      for ( int c = 0; c < olapCubes.size(); c++ ) {
+        if ( ( (OlapCube) olapCubes.get( c ) ).getName().equals( cubeName ) ) {
+          olapCube = (OlapCube) olapCubes.get( c );
+          break;
+        }
+      }
+      if ( olapCube == null ) {
+        throw new ModelerException(
+          BaseMessages.getString( "ModelAnnotation.resolveField.UNABLE_TO_FIND_CUBE", cubeName )
+        );
+      }
+
+      List usages = olapCube.getOlapDimensionUsages();
+      for ( int u = 0; u < usages.size(); u++ ) {
+        OlapDimensionUsage usage = (OlapDimensionUsage) usages.get( u );
+        OlapDimension olapDimension = usage.getOlapDimension();
+        List olapHierarchies = olapDimension.getHierarchies();
+        for ( int h = 0; h < olapHierarchies.size(); h++ ) {
+          StringBuffer buffer = new StringBuffer();
+          buffer.append( "[" );
+          buffer.append( usage.getName() );
+          OlapHierarchy olapHierarchy = (OlapHierarchy) olapHierarchies.get( h );
+          if ( StringUtils.isNotEmpty( olapHierarchy.getName() )
+            && !StringUtils.equals( olapHierarchy.getName(), usage.getName() ) ) {
+            buffer.append( "." ).append( olapHierarchy.getName() );
+          }
+          buffer.append( "].[" );
+          List hierarchyLevels = olapHierarchy.getHierarchyLevels();
+          for ( int hl = 0; hl < hierarchyLevels.size(); hl++ ) {
+            OlapHierarchyLevel olapHierarchyLevel = (OlapHierarchyLevel) hierarchyLevels.get( hl );
+            if ( levelName.equals( buffer.toString() + olapHierarchyLevel.getName() + "]" ) ) {
+              return olapHierarchyLevel.getReferenceColumn().getName( locale );
+            }
+          }
+        }
+      }
+      throw new ModelerException(
+        BaseMessages.getString( "ModelAnnotation.resolveField.UNABLE_TO_FIND_LEVEL", levelName )
+      );
+    } else {
+      throw new ModelerException(
```

If the else block just throws an exception, why not just test if the fields are blank and throw the exception immediately instead of nesting the rest of the method's logic